### PR TITLE
Merge To Master PR

### PR DESCRIPTION
**Pull request by othman-19**

### CODE REVIEW

**Comment by azdrenymeri** [comment]
## Hello @othman-19 and @wathigo 
### Good job on tests
* I see you have declared local variables and you need to use `let` to declare them
### Implement that change and add it for a re-review

File: spec/enumerable_spec.rb [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+               result.push("#{i} index \
+                     of #{x}")
+             end).to eql(arr)
+    end
+  end
+  #=end
+
+  #=begin
+  describe '#my_select' do
+    let(:arr) { [2, 4, 6, 8, 10, 12] }
```
Comment (mcrd25) [inline spec/enumerable_spec.rb]: This repeats several times in your tests... you can simply add it after line 7 and make it available to all subsequent tests...

You can do this for all let variable declarations

File: spec/enumerable_spec.rb [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+    it 'return a new array with asked condition in given block' do
+      expect(arr.my_select { |x| x > 6 }).to eql([8, 10, 12])
```
Comment (mcrd25) [inline spec/enumerable_spec.rb]: Don't forget you can use let to create blocks as well

**Review by mcrd25 — APPROVED** [review]
# Great Job! You may move on! 🚀 😄 

**While there are things that can be done to improve the way you did tests, it does its job and your methods work well! Please consider my review comments though and implement the change before merging to master! Congrats on completing the first part of your ruby testing!**

Great job! 💯 🥇 

***If you have any questions, now or in the future, do not hesitate to contact me on social media or on slack and I will be glad to chat with you.***
**[--Maya](https://github.com/mcrd25)**
[GitHub](https://github.com/mcrd25) [Twitter](https://twitter.com/mcrd25) [linkedin](https://www.linkedin.com/in/mayadouglas/)

**Pull request merged**